platform/source/mbed_retarget.cpp

```diff
@@ -516,8 +596,13 @@ extern "C" FILEHANDLE PREFIX(_open)(const char *name, int openflags)
     }
 #endif
     return open(name, openflags_to_posix(openflags));
+#else
+    // Everything goes to the same output
+    return STDOUT_FILENO;
```

Comment:
I think you should do something here to stop `fopen("Hello", "w")` from working.

And you do need to distinguish `STDIN_FILENO` and `STDOUT_FILENO` if the checks in `read` and `write` are going to work.

I think what you want for this function is to keep the top section, which is the special case for stdin/out/err, and only `#ifdef` out the "already-allocated" and "pass to `open`" bits  at the bottom - make it return -1 for anything other than stdin/out/err. (And set errno if we're bothering in minimal).

Reply:
> only #ifdef out the "already-allocated" and "pass to open" bits

The top bit references `mbed_file_handle` so I removed a bit from the top bit.
Please review [this](https://github.com/ARMmbed/mbed-os/compare/7b4d76b1df1bc516d678fcc17dd88d8b3a8794ea..046976a1a012e3703afcf64c177a410662d2bb93).

Thanks.

Reply:
Those `mbed_file_handle` calls were just there to force initialisation - make sure that relevant entries were filled in.

I suggest you could just make it return `nullptr` when minimal console, then that piece of code can be left in - you do need it to return the correct `xxx_FILENO` value.

Reply:
Do you mean not excluding `mbed_file_handle` but instead make it return `nullptr`?
If that is the case then it contradicts with the other thing we are trying to do (exclude `class FileHandle` in `mbed_retarget.h`) as the signature of the function is `FileHandle *mbed::mbed_file_handle(int fd)`.

Reply:
True, it would change that if it was retained publically.

But you could have it just defined internally for that code, and not declared publically in the header any more.

Not sure which is best - if it's not declared publically, then anyone using `mbed_file_handle(STDOUT_FILENO)` gets a compile error, and that may be preferable to having it compile and return NULL. Whatever they're trying to do won't work, and probably better to be loud about that at compile time.

Reply:
Please review [this](https://github.com/ARMmbed/mbed-os/compare/584fb82c08072bd67906b719416ab8257fe56bf0..af1ea588d5c9d649b9984b082397caf93f59022a).

Reply:
Fine apart from the declaration shifts.

You still need the `FileHandle` declaration in the header, inside the ifdef, for the functions in the header using it.

The C++ file doesn't a declaration, as it's always including "FilePath.h", which ends up giving you `FileHandle` (it's real definition, which you need). The C++ file can't just be relying on that mbed_retarget.h declaration.

Reply:
Thanks.
Please review [this](https://github.com/ARMmbed/mbed-os/compare/af1ea588d5c9d649b9984b082397caf93f59022a..e9869955fea7f8fa2f9781b81efda2a9868010fd).

Reply:
That seems correct now. Added a couple of notes in the surrounding code about the overrides.